backend/src/users/users.service.ts

```diff
@@ -59,4 +63,97 @@ export class UsersService {
     });
     return deleteUser;
   }
+
+  async addFriend(
+    userNickname: string,
+    friendNickname: string,
+    userId: number
+  ) {
+    const user = await this.getUser(userNickname);
+    const friend = await this.getUser(friendNickname);
+    if (!user || !friend) throw new NotFoundException('User not found');
+    if (userNickname === friendNickname || userId !== user.id) {
```

Comment:
Duplicate ?!
Can two users have the same nickname ?

Reply:
First one is for return error when user tries to add himself and second one is to verify if the user is the good user (a user can't add friend for another user)